### PR TITLE
Add rust-analyzer notes to quick-start.rst

### PR DESCRIPTION
**Pull request by adamrk**
Added a section in the quick start docs on how to configure rust-analyzer (from @wedsonaf's example https://github.com/Rust-for-Linux/linux/issues/202).

### CODE REVIEW

**Review by ojeda — COMMENTED** [review]
Probably it is best to put the JSON content directly in a `rust-project.json` file in `Doc/rust/`, so that people can quickly copy it to the root (after adjusting the sysroot).

Also, should we add it to the `.gitignore` (when in the root)?

File: Documentation/rust/quick-start.rst [hunk 1]
```diff
@@ -145,6 +145,24 @@ the component manually::
...
+
+``rust-analyzer`` will need to be
+`configured <https://rust-analyzer.github.io/manual.html#non-cargo-based-projects>`_
+to work with the kernel by adding a ``rust-project.json`` file in the root folder.
+The example ``Documentation/rust/rust-project.json`` can
+be used after updating ``sysroot_src`` and including the relevant modules.
+The path to ``sysroot_src`` is given by::
+
+    $(rustc --print sysroot)/lib/rustlib/src/rust/library
+
```
Comment (ojeda) [inline Documentation/rust/quick-start.rst:163]: Extra newline (I used 2 between sections).

**Review by ojeda — CHANGES_REQUESTED** [review]
Can you please sign the commit? (e.g. `git commit --signoff`). Since this is post-RFC, we will be following the kernel development process properly now (since we may or may not rebase later on).

File: Documentation/rust/quick-start.rst (hunk 1 above)
Comment (adamrk) [reply Documentation/rust/quick-start.rst:163]: Ah, I was thinking one after the code block and two additional...

**Pull request merged**